### PR TITLE
Prohibit renaming child trigger on distributed partition pre PG15

### LINKED ISSUE #6289: Misleading error when renaming child trigger on distributed partition
### Problem
Pre PG15, renaming child triggers on partitions is allowed. When creating a trigger in a distributed parent partitioned table, the triggers on the shards of the partitions have the same name with the triggers on the corresponding parent shards of the parent table. Therefore, they don't have the same appended shard id as the shard id of the partition. Hence, when trying to rename a child trigger on a partition of a distributed table, we can't correctly find the triggers on the shards of the partition in order to rename them since we append a different shard id to the name of the trigger. Since we can't find the trigger we get a misleading error of inexistent trigger.
### Steps to reproduce
```SQL
SET citus.enable_unsafe_triggers TO true;

CREATE TABLE sale(
    sale_date date not null,
    state_code text)
    PARTITION BY list (state_code);

ALTER TABLE sale ADD CONSTRAINT sale_pk PRIMARY KEY (state_code, sale_date);

CREATE TABLE sale_newyork PARTITION OF sale FOR VALUES IN ('NY');

SELECT create_distributed_table('sale', 'state_code');

CREATE FUNCTION funcA() RETURNS trigger AS $$
BEGIN
  RAISE NOTICE 'hello from funcA';
  RETURN null;
END; $$ LANGUAGE plpgsql;

CREATE TRIGGER my_trig
  AFTER INSERT ON sale
  FOR EACH ROW EXECUTE FUNCTION funcA();

ALTER TRIGGER my_trig ON sale_newyork RENAME TO new_trig_name;
```

### Error we face
```
ERROR:  trigger "my_trig_102040" for table "sale_newyork_102040" does not exist
CONTEXT:  while executing command on <worker>
```

### PR DESCRIPTION
**Pull request by naisila**
Pre PG15, renaming child triggers on partitions is allowed. When creating a trigger in a distributed parent partitioned table, the triggers on the shards of the partitions have the same name with the triggers on the corresponding parent shards of the parent table. Therefore, they don't have the same appended shard id as the shard id of the partition. Hence, when trying to rename a child trigger on a partition of a distributed table, we can't correctly find the triggers on the shards of the partition in order to rename them since we append a different shard id to the name of the trigger. Since we can't find the trigger we get a misleading error of inexistent trigger.

In this PR we prohibit renaming child triggers on distributed partitions altogether.

Fixes #6289

### CODE REVIEW

File: src/backend/distributed/commands/trigger.c [hunk 1]
```diff
@@ -611,6 +648,64 @@ ErrorOutForTriggerIfNotSupported(Oid relationId)
...
 
 
+#if (PG_VERSION_NUM < PG_VERSION_15)
+
+/*
+ * ErrorOutIfCloneTrigger is a helper function to error
+ * out if we are trying to rename a child trigger on a
+ * partition of a distributed table.
+ * A lot of this code is borrowed from PG15 because
+ * renaming clone triggers isn't allowed in PG15 anymore.
```
Comment (marcocitus) [inline src/backend/distributed/commands/trigger.c:658]: if it's not supported, couldn't we just let it error out in ProcessUtility or do we run into trouble before then?

File: src/test/regress/expected/upgrade_distributed_triggers_before.out [hunk 2]
```diff
@@ -241,3 +241,6 @@ SELECT * FROM sale_triggers ORDER BY 1, 2;
  truncate_trigger_xxxxxxx      | sale_california | O
 (12 rows)
 
+-- check that we can't rename child triggers on partitions of distributed tables
+ALTER TRIGGER another_trigger ON sale_newyork RENAME TO another_renamed_trigger;
+ERROR:  cannot rename child triggers on distributed partitions
```
Comment (hanefi) [inline src/test/regress/expected/upgrade_distributed_triggers_before.out:246]: This error message will make it harder for us to add PG16 support in the future. As this error happens on versions before PG15, this test will fail on PG15->PG16 upgrades.

How about adding the test in the regression suite in a new test file that will be run only for PG versions earlier than 15?

File: src/backend/distributed/commands/trigger.c (hunk 1 above)
Comment (hanefi) [reply src/backend/distributed/commands/trigger.c:658]: > couldn't we just let it error out in ProcessUtility?

For PG15 yes. For earlier versions, no as the error message is confusing and misleading.

Renaming child triggers is prohibited on PG15 only. The change is not backported to earlier versions of PG.

This helper function is only used in earlier versions of PG, where renaming of child triggers are still allowed. The rename works on coordinator shell table, but gives misleading error messages as the triggers on the shards are not found with the expected names.

Comment (marcocitus) [reply src/backend/distributed/commands/trigger.c:658]: got it, thanks

File: src/test/regress/expected/upgrade_distributed_triggers_before.out (hunk 2 above)
Comment (naisila) [reply src/test/regress/expected/upgrade_distributed_triggers_before.out:246]: Actually this test is only run in PG14 so we're good.

**Review by hanefi — APPROVED** [review]

**Pull request merged**